### ISSUE TITLE
Executor step with different world_size

### ISSUE THREAD

**Issue by Derek-Gong** [wenet-e2e/wenet#622]
I just found that executor.py does not consider world_size and results that different # of GPUs leads to different # of steps in one epoch. 
So, we get different LR for the same epoch using different # of GPUs, which is not what I expect when reproducing experiments.
Another more realistic side-effect is that with more GPUs LR changes slower because # of the batch is reduced on each GPU!

I've tried to fix that by adding an extra parameter "world_size" to executor.train, 
and two lines at https://github.com/wenet-e2e/wenet/blob/15ccd07de4ff9c01824de92b9ef721955068e686/wenet/utils/executor.py#L86-L87:

[scheduler.step() for i in range(world_size)]
self.step += world_size

I wonder if there's a better solution for this issue.

**Comment by robin1001** [comment]
Pytorch use synchronous training for DDP, which means we should have same step number for every GPU.

**Comment by Derek-Gong** [comment]
That's true. But this is not about synchronization. I would also like to keep the LR schedule the same no matter how many GPUs I use. For now, If I increase the number of GPUs, LR increasing would be slower while the warm-up scheduler is used. Somehow, that behavior cancels the effect of using more GPUs.
